src/app/shared/provider.service.ts

```diff
@@ -0,0 +1,61 @@
+export class ProviderService {
+
+  /* set up project provider */
+  setUpProvider(tool) {
+    const gitUrl = tool.gitUrl;
+
+    tool.provider = this.getProvider(gitUrl);
+    tool.providerUrl = this.getProviderUrl(gitUrl, tool.provider);
+
+    return tool;
+  }
+
+  private getProvider(gitUrl: string): string {
+    if (gitUrl.includes('github.com')) {
```

Comment:
Add a todo here. Without an anchor, this looks fragile (for example if you had a github repo that included the string " bitbucket.org" in its name.